mmedit/models/editors/stable_diffusion/clip_wrapper.py

```diff
@@ -0,0 +1,173 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import os
+
+import torch
+import torch.nn as nn
+from mmengine.logging import MMLogger
+from transformers import CLIPConfig, CLIPVisionModel, PreTrainedModel
+from transformers.models.clip.feature_extraction_clip import \
+    CLIPFeatureExtractor  # noqa
+from transformers.models.clip.modeling_clip import CLIPTextModel
+from transformers.models.clip.tokenization_clip import CLIPTokenizer
```

Comment:
We should use `try_import` here since `transformers` is an optional package.

e.g.,
```python
from mmediting import try_import
transformers = try_import('transformers')

if transformers is None:
        from transformers import CLIPConfig, CLIPVisionModel, PreTrainedModel
    from transformers.models.clip.feature_extraction_clip import \
        CLIPFeatureExtractor  # noqa
    from transformers.models.clip.modeling_clip import CLIPTextModel
    from transformers.models.clip.tokenization_clip import CLIPTokenizer

    class StableDiffusionSafetyChecker(PreTrainedModel):
        ...

    def load_clip_submodels(init_cfg, submodules, requires_safety_checker):
        ...

else:
    def load_clip_submodels(*args, **kwargs):
        raise ImportError('Please install tranformers via '
                          '\'pip install transformers\'')

```

Reply:
Then we can use the following unit test to cover these modified lines:
```python
import pytest
import sys


def test_transformers_is_none():
    sys.modules['transformers'] = None
    from new_clip_wrapper import load_clip_submodels

    with pytest.raises(ImportError):
        load_clip_submodels('just a test')


def test_wrapper():
    ...

```

Reply:
Then we can use the following unit test to cover these modified lines:
```python
import pytest
import sys


def test_transformers_is_none():
    sys.modules['transformers'] = None
    from new_clip_wrapper import load_clip_submodels

    with pytest.raises(ImportError):
        load_clip_submodels('just a test')


def test_wrapper():
    ...

```

Reply:
resolved